realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoCollection.java

```diff
@@ -62,7 +66,30 @@
      * @return the namespace
      */
     public MongoNamespace getNamespace() {
-        throw new RuntimeException("Not Implemented");
+        return nameSpace;
+    }
+
+    /**
+     * Get the class of documents stored in this collection.
+     * <p>
+     * If you used the simple {@link MongoDatabase#getCollection(String)} to get
+     * this collection,
+     * this is {@link org.bson.Document}.
+     * </p>
+     *
+     * @return the class
+     */
+    public Class<DocumentT> getDocumentClass() {
+        return osMongoCollection.getDocumentClass();
+    }
+
+    /**
+     * Get the codec registry for the RemoteMongoCollection.
+     *
+     * @return the {@link CodecRegistry}
```

Comment:
This kind of JavaDoc doesn't really say anything the types dont already. I would probably rephrase to `@return the {@link CodecRegistry}  for this collection`.